### ISSUE TITLE
menu-bar: Focus is lost after pressing "esc" while being in nested menu

### ISSUE THREAD

**Issue by vtknann** [angular/material#11678]
## Bug:
If you pressing "ecs" while being in a nested menu, focus is lost. This is bad for accessibility users, which rely on focus heavily to navigate. Instead focus should be returned on previous menu item or on the menu button. 

#### Detailed Reproduction Steps: 
1. Go to [Menu Bar basic usage demo](https://material.angularjs.org/latest/demo/menuBar)
1. Using the keyboard, navigate to File > New >  Presentation
![Screenshot from 2019-03-14 14-07-27](https://user-images.githubusercontent.com/4023939/54359260-9742fd80-4662-11e9-864c-06568942c0da.png)
1. Press "ecs". 
1. Focus is lost.
### What is the expected behavior?
[WAI-ARIA menu practices for accessibility](https://www.w3.org/TR/wai-aria-practices/#menu) states that on
Escape: Close the menu that contains focus and return focus to the element or context, e.g., menu button or parent menuitem, from which the menu was opened.

### What is the current behavior?
Focus is lost.

### What is the use-case or motivation for changing an existing behavior?
Accessibility users (screen reader users, keyboard users, switch users) use focus heavily to navigate. Then focus is suddenly lost it's a frustrating experience. 

### Which versions of AngularJS, Material, OS, and browsers are affected?
- AngularJS Material: Latest Release 1.1.13
- Browsers: Chrome

**Comment by Splaktar** [comment]
Thank you for reporting this! This certainly needs to be improved.

**Comment by marosoft** [comment]
@Splaktar I have a fix but I am working on the test. There are none for any keyboard interactions except Esc both in Menu and Menu Bar. This one is wrong for the nested menu.
